chalk/transform.py

```diff
@@ -47,123 +208,166 @@ def __rmul__(self, t: Affine) -> Self:
 
-    def rotate(self, θ: float) -> Self:
+    def rotate(self, θ: Floating) -> Self:
         "Rotate by θ degrees counterclockwise"
-        return self._app(Affine.rotation(θ))
+        return self._app(rotation(to_radians(θ)))
 
-    def rotate_rad(self, θ: float) -> Self:
+    def rotate_rad(self, θ: Floating) -> Self:
         "Rotate by θ radians counterclockwise"
-        return self._app(Affine.rotation(from_radians(θ)))
+        return self._app(rotation((θ)))
 
-    def rotate_by(self, turns: float) -> Self:
+    def rotate_by(self, turns: Floating) -> Self:
         "Rotate by fractions of a circle (turn)."
         θ = 2 * math.pi * turns
-        return self._app(Affine.rotation(from_radians(θ)))
+        return self._app(rotation((θ)))
 
     def reflect_x(self) -> Self:
-        return self._app(Affine.scale(Vec2(-1, +1)))
+        return self._app(scale(V2(-1, +1)))
 
     def reflect_y(self) -> Self:
-        return self._app(Affine.scale(Vec2(+1, -1)))
+        return self._app(scale(V2(+1, -1)))
 
-    def shear_y(self, λ: float) -> Self:
-        return self._app(Affine(1.0, 0.0, 0.0, λ, 1.0, 0.0))
+    def shear_y(self, λ: Floating) -> Self:
+        return self._app(make_affine(1.0, 0.0, 0.0, λ, 1.0, 0.0))
 
-    def shear_x(self, λ: float) -> Self:
-        return self._app(Affine(1.0, λ, 0.0, 0.0, 1.0, 0.0))
+    def shear_x(self, λ: Floating) -> Self:
+        return self._app(make_affine(1.0, λ, 0.0, 0.0, 1.0, 0.0))
 
-    def translate(self, dx: float, dy: float) -> Self:
-        return self._app(Affine.translation(Vec2(dx, dy)))
+    def translate(self, dx: Floating, dy: Floating) -> Self:
+        return self._app(translation(V2(dx, dy)))
 
     def translate_by(self, vector) -> Self:  # type: ignore
-        return self._app(Affine.translation(vector))
-
+        return self._app(translation(vector))
 
-# Below here are a collection of hacks to ensure that planar objects
-# behave like the rest of the Chalk library. We do this by monkey
-# patching in methods to Vec2 and by fixing a bug in the Affine
-# transformation. This is not great, but necessary to keep the
-# Object oriented api of Chalk.
+@dataclass
+class BoundingBox(Transformable):
+    tl: P2_t
+    br: P2_t
 
-Vec2._app = lambda x, y: y * x  # type: ignore
-Vec2.shear_x = Transformable.shear_x  # type: ignore
-Vec2.shear_y = Transformable.shear_y  # type: ignore
-Vec2.scale = Transformable.scale  # type: ignore
-Vec2.scale_x = Transformable.scale_x  # type: ignore
-Vec2.scale_y = Transformable.scale_y  # type: ignore
-Vec2.rotate = Transformable.rotate  # type: ignore
-Vec2.rotate_by = Transformable.rotate_by  # type: ignore
-Vec2.reflect_x = Transformable.reflect_x  # type: ignore
-Vec2.reflect_y = Transformable.reflect_y  # type: ignore
-V2 = Vec2
+    def apply_transform(self, t: Affine) -> Self:
+        tl = t @ self.tl
+        br = t @ self.br
+        tl2 = np.minimum(tl, br)
+        br2 = np.maximum(tl, br)
```

Comment:
To compute the transformed bounding box we need to consider all four corners not just the two opposite ones. For example, rotating a box by 45 degrees will result in a thin line if we were to look only at the top-left and bottom-right corners.